pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -223,6 +238,15 @@ func (r *ReconcileWildFlyServer) Reconcile(request reconcile.Request) (reconcile
 		}
 	}
 
+	// create a Prometheus ServiceMonitor (if the resource exists on the cluster)
+	if hasServiceMonitor() {
```

Comment:
Should not be include the case when the ServiceMonitor resource was deleted on the cluster? If so, I would expect to delete any resource created in a previous loop.

In addition to that, is the service monitor resource created for WildFly deleted when the WildFly Operator is removed? ... I'm not sure if that is happening

Reply:
The created ServiceMonitor is owned by the WildFlyServer resource. It will be deleted when the WildFlyServer resource itself is deleted.

If the ServiceMonitor CRD is removed from the cluster, the one created by the WildFlyServer will also be removed. When the reconciler loop is run, we will not create a new one as the CRD is no longer present.

Regarding the removal of the operator: when the operator is removed, it does not impact the resources that were deployed. The `WildFlyServer` CRD will still be available.
It's only when you delete the `WildFlyServer` CRD that you would delete the corresponding resource and any other resources that are owned by it (including its ServiceMonitor)